src/Model/Configuration.php

```diff
@@ -31,36 +31,59 @@ class Configuration
     protected $composerConfiguration;
     /** @var PackageConfigurationAbstract $packageConfiguration */
     protected $packageConfiguration;
-
+    /** @var Parser $parser */
+    protected $parser;
 
     /**
      * Configuration constructor.
      *
      * @param Parser $parser
      *
+     * @since 0.0.1
+     */
+    public function __construct(Parser $parser)
+    {
+       $this->setParser($parser);
+    }
+
+    /**
+     * Initialize configurations
+     *
+     * @return $this
      * @throws MissingConfigurationException
      * @throws MissingParameterException
+     *
+     * @since 0.0.1
      */
-    public function __construct(Parser $parser)
+    public function init()
     {
-        $this->configuration = $parser->parse();
+        $this->setConfiguration();
         $this->initPackageConfiguration();
         $this->initComposerConfiguration();
+
+        return $this;
     }
 
     /**
-     * Fetch the given key from the loaded configuration
+     * Set configuration property
      *
-     * @param string|int $key
-     * @param bool $mandatory Throw error if value does not exist
+     * @since 0.0.1
+     */
+    protected function setConfiguration()
```

Comment:
maybe more context - we parse a file within this setter... or setConfiguration should be called with configuration content to ensure single responsibility?